torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -2749,6 +2763,34 @@ def _test_all_gather_helper(
 
             self._barrier()
 
+        def test_all_gather_quant(
```

Comment:
Can't we just reuse `_test_all_gather_helper` here and add a `quant` parameter to `_test_all_gather_helper` which calls `auto_quantize`? That way we don't repeat a lot of the logic here.